src/nvim/ex_cmds.c

```diff
@@ -5046,6 +5008,46 @@ int find_help_tags(char_u *arg, int *num_matches, char_u ***matches, int keep_la
   return OK;
 }
 
+// Called when starting to edit a buffer for a help file.
+static void prepare_help_buffer(void)
```

Comment:
We autogenerate the header files, so we don't really have the distinction between declaration/definition comments and the `.c` files have all the comments. See e.g. [`buf_jump_open_win` in `src/nvim/window.c`](https://github.com/neovim/neovim/blob/master/src/nvim/window.c#L3567) and its [docu](https://neovim.org/doc/dev/window_8c.html#a0f972066bca0dc1d7b181e54ff6dc038). So I still think this should be changed.

I guess the style guide should be updated..